src/components/ListBox/component.js

```diff
@@ -0,0 +1,55 @@
+import { h, Fragment } from "preact";
+import {
+  ListboxInput,
+  ListboxButton,
+  ListboxPopover,
+  ListboxList,
+  ListboxOption,
+} from "@reach/listbox";
+import "@reach/listbox/styles.css";
+import { Icon } from "@Components/Icon";
+import PropTypes from 'prop-types';
+
+const ListBox = ({ defaultValue, placeholder, options, onChange }) => {
+  /**
+   * If no options yet, we don't want to render the component.
+   */
+  if (!options) {
+    return;
+  }
```

Comment:
When would you not set options on this component? Should this not be a responsibility of the caller instead?

Reply:
If you are using this component as a `Web Component`, you would do something like:

```html
<sn-listbox id="my_listbox"></sn-listbox>
```
You are not setting options initially (because you don't have them yet, for example). You then initialize options via Javascript when you are ready:

```javascript
const myListbox = document.getElementByID("my_listbox");
myListbox.options = [
...
];
```

---

The below block of code is simply to prevent rendering the component without options:
```javascript
...
if (!options) {
    return;
}
...
```

Without this, an error is thrown that the `options` property was not set. This means that you can use it after the error is thrown.

Reply:
I was thinking options should always exist, even as an empty array. But I admit it's more of a taste thing :) all good then.